ethpm_cli/install.py

```diff
@@ -47,11 +34,97 @@ def install_package(pkg: Package, config: Config) -> None:
+    # Copy temp package directory to ethpm dir namespace
     dest_pkg_dir = config.ethpm_dir / pkg.alias
     validate_parent_directory(config.ethpm_dir, dest_pkg_dir)
     shutil.copytree(tmp_pkg_dir, dest_pkg_dir)
-    update_ethpm_lock(pkg, (config.ethpm_dir / "ethpm.lock"))
+    install_to_ethpm_lock(pkg, (config.ethpm_dir / "ethpm.lock"))
+
+
+_InstalledPackageTree = namedtuple(
+    "InstalledPackageTree", "depth path manifest children content_hash"
+)
+
+
+class InstalledPackageTree(_InstalledPackageTree):
+    depth: int
+    path: Path
+    manifest: Dict[str, Any]
+    children: Tuple["InstalledPackageTree", ...]
+    content_hash: str
+
+    @property
+    def package_name(self) -> str:
+        return self.manifest["package_name"]
+
+    @property
+    def package_version(self) -> str:
+        return self.manifest["version"]
+
+    @property
+    def format_for_display(self) -> str:
+        prefix = "- " * self.depth
+        columns = get_terminal_width()
+        main_info = f"{prefix}{self.package_name}=={self.package_version}..."
+        hash_info = f"({self.content_hash})"
+        diff = columns - len(main_info) - len(hash_info)
+        if self.children:
+            children = "\n" + "\n".join(
+                (child.format_for_display for child in self.children)
+            )
+        else:
+            children = ""
+        return f"{main_info}{'.' * diff}{hash_info}{children}"
+
+
+def list_installed_packages(config: Config) -> None:
+    installed_packages = [
+        get_installed_package_tree(base_dir)
+        for base_dir in config.ethpm_dir.glob("*/")
+        if base_dir.is_dir()
+    ]
+    for pkg in sorted(installed_packages):
+        logger.info(pkg.format_for_display)
+
+
+def get_installed_package_tree(base_dir: Path, depth: int = 0) -> InstalledPackageTree:
+    manifest = json.loads((base_dir / "manifest.json").read_text())
+    ethpm_lock = json.loads((base_dir.parent / "ethpm.lock").read_text())
+    content_hash = ethpm_lock[base_dir.name]["resolved_uri"]
+    dependency_dirs = get_dependency_dirs(base_dir)
+    children = tuple(
+        get_installed_package_tree(dependency_dir, depth + 1)
+        for dependency_dir in dependency_dirs
+    )
+    return InstalledPackageTree(depth, base_dir, manifest, children, content_hash)
+
+
+@to_list
+def get_dependency_dirs(base_dir: Path) -> Iterable[Path]:
+    dep_dir = base_dir / "ethpm_packages"
+    if dep_dir.is_dir():
+        for ddir in dep_dir.glob("*/"):
+            if ddir.is_dir() and ddir.name != "src":
+                yield ddir
+
+
+def is_package_installed(package_name: str, config: Config) -> bool:
+    return os.path.exists(config.ethpm_dir / package_name)
+
+
+def uninstall_package(package_name: str, config: Config) -> None:
+    if not is_package_installed(package_name, config):
+        raise InstallError(
+            f"Unable to uninstall {package_name} from {config.ethpm_dir}"
```

Comment:
This message would confuse me.  I read this as "the package is installed but I can't uninstall it".   Maybe less ambiguous messaging would be: "No package with the name {package_name} found installed under {config.ethpm_dir}"